exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer.go

```diff
@@ -106,12 +110,44 @@ func appendMetricWithParentEntity(helixMetrics []BMCHelixOMMetric, helixMetric B
 	return append(helixMetrics, helixMetric)
 }
 
-// createHelixMetric converts a single OpenTelemetry metric into a BMCHelixOMMetric payload
-func (mp *MetricsProducer) createHelixMetric(metric pmetric.Metric, resourceAttrs map[string]string) (*BMCHelixOMMetric, error) {
+// createHelixMetrics converts each OpenTelemetry datapoint into an individual BMCHelixOMMetric
+func (mp *MetricsProducer) createHelixMetrics(metric pmetric.Metric, resourceAttrs map[string]string) ([]BMCHelixOMMetric, error) {
+	var helixMetrics []BMCHelixOMMetric
+
+	switch metric.Type() {
+	case pmetric.MetricTypeSum:
+		for i := 0; i < metric.Sum().DataPoints().Len(); i++ {
+			dp := metric.Sum().DataPoints().At(i)
+			metricPayload, err := mp.createSingleDatapointMetric(dp, metric, resourceAttrs)
+			if err != nil {
+				mp.logger.Warn("Failed to create Helix metric from datapoint", zap.Error(err))
+				continue
+			}
+			helixMetrics = append(helixMetrics, *metricPayload)
+		}
+	case pmetric.MetricTypeGauge:
+		for i := 0; i < metric.Gauge().DataPoints().Len(); i++ {
+			dp := metric.Gauge().DataPoints().At(i)
+			metricPayload, err := mp.createSingleDatapointMetric(dp, metric, resourceAttrs)
+			if err != nil {
+				mp.logger.Warn("Failed to create Helix metric from datapoint", zap.Error(err))
+				continue
+			}
+			helixMetrics = append(helixMetrics, *metricPayload)
+		}
+	default:
+		return nil, fmt.Errorf("unsupported metric type %s", metric.Type())
```

Comment:
Maybe a "warning" is enough for this. It's normal to discard unsupported metrics.

Reply:
Yes, `Errorf` formats the message using a format specifier and returns it as a value that implements the `error` interface (similar to an exception). In this context, the caller logs it as a warning, so this behavior should be good.